include/xtensor/xbuffer_adaptor.hpp

```diff
@@ -1092,13 +1147,25 @@ namespace xt
     }
 
     template <class I, class CI>
-    inline auto xiterator_adaptor<I, CI>::data() noexcept -> iterator
+    inline auto xiterator_adaptor<I, CI>::data() noexcept -> pointer
+    {
+        return &(*m_it);
```

Comment:
This is incorrect for iterator returning rvalues when dereferenced. This should be reverted to the previous implementation, except if this was leading to a bug. In that case, `xtl::xclosure_wrapper / xtl::xclosure_pointer` should be used instead (but I'm really not sure we need them here).